GetStream/stream-chat-swift
# [CIS-1885] Support ephemeral messages on MockServer

### 🔗 Issue Links

- https://stream-io.atlassian.net/browse/CIS-1885

### 🎯 Goal

- To support sending/receiving ephemeral messages via MockServer

### 📝 Summary

`UserRobot` now can:
- `sendGiphy()`
- `replyWithGiphy()`
- `replyWithGiphyInThread()`
- `tapOnSendGiphyButton()`
- `tapOnShuffleGiphyButton()`
- `tapOnCancelGiphyButton()`

`ParticipantRobot` now can:
- `sendGiphy()`
- `replyWithGiphy()`
- `replyWithGiphyInThread()`

`MockServer` now:
- saves removed messages
- saves messages after adding reactions
- replaces updated messages instead of saving them as new ones
- does not use `websocketDelay`

### 🎨 Showcase

https://user-images.githubusercontent.com/20794991/170738899-b30b3355-b3db-4e7c-a1d6-d0c1f751dfec.mp4

### 🧪 Manual Testing Notes

- Run a test app and play with ephemeral messages and everything around them (e.g.: reactions)
- Create a new automation test and check out the new methods

### ☑️ Contributor Checklist

### 🎁 Meme

![gif](https://media.giphy.com/media/3oEjI4PRkwjFqnd7CE/giphy.gif)
